### PR TITLE
Add `allowinsecureconnections` option

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
## Bug

Fixes: https://github.com/NuGet/Home/issues/13378

Regression? Last working version:

## Description

`nuget add` and `nuget update` commands do not allow for adding a new package source with an attribute `allowInsecureConnections`.  Users would have to go to the config file manually to add this attribute. 
* This PR adds a `-allowInsecureConeections` option to these commands.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [x] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  https://github.com/NuGet/Home/issues/13498 , https://github.com/dotnet/docs/issues/40488
  - [x] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx [hunk 1]
```diff
@@ -931,4 +931,8 @@ nuget client-certs List</value>
   <data name="SourcesCommandProtocolVersionDescription" xml:space="preserve">
     <value>The NuGet server protocol version to be used. Currently supported versions are 2 and 3. See https://learn.microsoft.com/nuget/api/overview for information about the version 3 protocol. Defaults to 2 if not specified.</value>
   </data>
+  <data name="SourcesCommandAllowInsecureConnectionsDescription" xml:space="preserve">
+    <value>Enables HTTP connections, applicable only to the `Add` and `Update` subcommands. Note: Data sent or received is not secure. Please refer to https://aka.ms/nuget-https-everywhere.</value>
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx]: Is the `Data sent or received is not secure` supposed to apply to the https everywhere?

**Review by nkolev92 — COMMENTED** [review]
LGTM! Thanks for focusing on a complete experience.
 
* 1 question about the string.
* Can you please file a documentation issue for this addition. This needs to capture dotnet.exe and nuget.exe and the dotnet.exe docs are in the dotnet/docs repo.

File: src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx (hunk 1 above)
Comment (Nigusu-Allehu) [reply src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx]: It is to discourage http usage and make sure users know it is not secure. We could have `Enables HTTP connections, applicable only to the `Add` and `Update` subcommands. Note: HTTP connections are insecure. Please refer to https://aka.ms/nuget-https-everywhere.`

Comment (jeffkl) [inline src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx]: Some users have reported links not working if the message ends in a period.  We might want to omit it in this case or add more words at the end like "for more information."

Comment (Nigusu-Allehu) [reply src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx]: @JonDouglas  any suggestions?

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 2]
```diff
@@ -912,4 +912,8 @@ Non-HTTPS access will be removed in a future version. Consider migrating to 'HTT
   <data name="ConfigGetCommandDescription" xml:space="preserve">
     <value>Gets the NuGet configuration settings that will be applied.</value>
   </data>
-</root>
+  <data name="SourcesCommandAllowInsecureConnectionsDescription" xml:space="preserve">
+    <value>Enables HTTP connections, applicable only to the `Add` and `Update` subcommands. Note: Data sent or received is not secure. Please refer to https://aka.ms/nuget-https-everywhere.</value>
```
Comment (JonDouglas) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]:
```suggestion
    <value>Allows HTTP connections for adding or updating packages. Note: This method is not secure. For secure options, see https://aka.ms/nuget-https-everywhere.</value>
```

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Verbs.cs [hunk 3]
```diff
@@ -390,6 +395,10 @@ internal partial class UpdateVerbParser
                         "--configfile",
                         Strings.Option_ConfigFile,
                         CommandOptionType.SingleValue);
+                    CommandOption allowInsecureConnections = SourceCmd.Option(
+                        "--allowInsecureConnections",
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Verbs.cs:399]: Options in dotnet are all lower case and snake case. 

See: store-password-in-clear-text, valid-authentication-types etc.

**Review by nkolev92 — APPROVED** [review]
Nicely done.

**Pull request merged**